Dockerfile

```diff
@@ -1,33 +1,106 @@
-FROM node:20.2.0-slim
+FROM node:20.2.0-slim as compiler
+
+#RUN apt-get update && apt-get -y install git wget openssl
+
 WORKDIR /app
 
-# Define node.js environment variables
+#RUN git clone https://github.com/ajnart/homarr.git .
+COPY . .
+
+RUN yarn install
+COPY .env.example .env
+RUN yarn build
+
+
+FROM node:20.2.0-alpine3.18
+
+#ARGS is only for build
+
 ARG PORT=7575
 
+# Keep free id >= 1000 for user, under node:x image by default node user uses 1000:1000
+ARG NODE_UID=800
+ARG NODE_GID=800
+
+#PUID can be set during build and run time
+ARG PUID=801
+ARG PGID=801
+
+#it must be the same as the host, temporary 802 or any, automatically changed at runtime
+ARG DOCKER_GID=802
+
+#By default, ping group using gid 999, keep free to possible docker host gid
+ARG PING_GID=803 
+
+# Expose the default application port
+EXPOSE $PORT
+ENV PORT=${PORT}
+
+# Define node.js environment variables
 ENV NEXT_TELEMETRY_DISABLED 1
 ENV NODE_ENV production
 ENV NODE_OPTIONS '--no-experimental-fetch'
 
-COPY next.config.js ./
-COPY public ./public
-COPY package.json ./temp_package.json
-COPY yarn.lock ./temp_yarn.lock
+# App environment variables
+ENV DATABASE_URL "file:/app/data/db.sqlite"
+ENV NEXTAUTH_URL "http://localhost:7575"
+ENV NEXTAUTH_SECRET NOT_IN_USE_BECAUSE_JWTS_ARE_UNUSED
+
+# Must be same as host user when using bind mount volumes
+ENV PUID $PUID
+ENV PGID $PGID
+
+RUN apk update && apk add --no-cache \
+    supervisor docker-cli shadow
+
+RUN usermod -u $NODE_UID node
+RUN groupmod -g $NODE_GID node
+
+RUN groupmod -g $PING_GID ping
+
+# Creating local homarr user and group
+RUN groupadd -g $PGID homarr
+RUN useradd homarr -u $PUID -g homarr --home-dir /app --shell /sbin/nologin
+RUN usermod -aG node homarr
+
+# Creating a local Docker group and add docker group to homarr user
+RUN groupadd -g $DOCKER_GID docker
+RUN usermod -aG docker homarr
```

Comment:
This among other things is quite concerning. I will respect the projects security policy and avoid going into detail here.